### PR TITLE
PMP isotropic remeshing : avoid creating self intersections near patch boundaries

### PR DESCRIPTION
**Pull request by janetournois**
the collapse step was still creating self-intersections
when it happened in a particular configuration close to patch boundaries

### CODE REVIEW

**Comment by janetournois** [comment]
this PR is not ready for testing

**Comment by janetournois** [comment]
ready for testing!

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h [hunk 1]
```diff
@@ -1116,6 +1128,48 @@ namespace internal {
...
+        { 
+          CGAL_assertion(patch_ids_map_[nxt] == patch_ids_map_[face(h, mesh_));
+          return nxt;
+        }
+        nxt = next(opposite(nxt, mesh_), mesh_);
+      }
+      while (end != nxt);
+
+      CGAL_assertion(false);
+      return end;
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h]: `return end`?????

File: Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h [hunk 2]
```diff
@@ -1116,6 +1128,48 @@ namespace internal {
       return true;//we already checked we're not pinching a hole in the patch
     }
 
+    halfedge_descriptor next_on_patch_border(const halfedge_descriptor& h) const
+    {
+      CGAL_precondition(is_on_patch_border(h));
+
+      halfedge_descriptor end = h;
```
Comment (sloriot) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h]: `end = opposite(h,mesh_)`?

Comment (janetournois) [inline Polygon_mesh_processing/include/CGAL/Polygon_mesh_processing/internal/Isotropic_remeshing/remesh_impl.h]: absolutely! I'll fix it!

**Pull request closed without merging**